platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/offline/MergeOfflineRegionsActivity.kt

```diff
@@ -57,6 +58,23 @@ class MergeOfflineRegionsActivity : AppCompatActivity() {
     }
   }
 
+  /**
+   * Since we expect from the results of the offline merge callback to interact with the hosting activity,
+   * we need to ensure that we are not interacting with a destroyed activity.
+   */
+  private class MergeCallback(activityCallback: OfflineManager.MergeOfflineRegionsCallback) : OfflineManager.MergeOfflineRegionsCallback {
+
+    private val activityWeakReference = WeakReference<OfflineManager.MergeOfflineRegionsCallback>(activityCallback)
+
+    override fun onMerge(offlineRegions: Array<out OfflineRegion>?) {
+      activityWeakReference.get()?.onMerge(offlineRegions)
```

Comment:
What happens if this callback is called between onDestroy and when the activity is cleared from memory? It could be a couple of seconds until GC kicks in, clears the activity, and then finds and clears the WeakReference above. During this time you are trying to set styles and merge regions but the UI will not allow it, and it could lead to crashes like the classic `Can not perform this action after onSaveInstanceState`.

Reply:
Thanks @pakoito. In this case, since the callback interacts with the map, it's definitely cleaner to make sure to release the reference proactively in the `#onDestroy` instead of waiting for the GC.

Reply:
Gotcha, np! If there are no other comments/concerns I'm going to go ahead and start the work on the remaining to-do items.

Reply:
> Gotcha, np! If there are no other comments/concerns I'm going to go ahead and start the work on the remaining to-do items.

Sounds good to me @LukasPaczos I don't know how this thread got out of order 🤔 almost missed ☝️ comment 😅

Reply:
Github had a hard day yesterday, all is in order now :)